src/asciidoc/docs/develop/testingTemplates.adoc

```diff
@@ -0,0 +1,171 @@
+= Testing templates
+:page-slug: testing-templates
+:page-description: Guide to testing Koji templates.
+
+There are many powerful testing tools built into the Koji platform.
+Typically, you will begin testing in the Koji editor.
+Later, you'll want to publish the template as *Unlisted*, so you can test it in a production environment before giving access to general users.
+Some of the most common testing methods are listed below.
+
+== Testing in the Koji editor
+
+The Koji editor provides various ways to preview a template's behavior.
+
+=== Previewing the Koji
+
+The *Live preview* tab allows you to toggle between *Preview* mode, which shows what the Koji looks like when running, and *Editing* mode, which allows you to perform remixing in the preview.
+
+The *Remote* tab offers several additional ways of previewing your Koji.
+
+* You can test it on your mobile device by scanning a QR code.
```

Comment:
These bullets are much improved from the previous version. One suggestion for simplifying the phasing... What do you think of opening each bullet with a short summary of the action, rather than a full sentence? Eg.

- On a mobile device – Generate a QR code that you can scan from any device. Logs from...
- In a new tab – Open a preview on a staging server. Log...
- In the live player – Text.

Reply:
My only problem with this is that the user doesn't actively generate the QR code -- it's there as soon as he/she opens the tab. So maybe change "Generate" to "Generates" or reword the phrase as follows: "Open a preview on any device by scanning a QR code." What do you think?

Reply:
Yes, I like your phrasing!